core/src/main/java/org/elasticsearch/index/query/FieldMaskingSpanQueryBuilder.java

```diff
@@ -19,27 +19,59 @@
 
 package org.elasticsearch.index.query;
 
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.spans.FieldMaskingSpanQuery;
+import org.apache.lucene.search.spans.SpanQuery;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
 import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.index.mapper.MappedFieldType;
 
 import java.io.IOException;
+import java.util.Objects;
 
-public class FieldMaskingSpanQueryBuilder extends QueryBuilder implements SpanQueryBuilder, BoostableQueryBuilder<FieldMaskingSpanQueryBuilder> {
+/**
+ *
+ */
+public class FieldMaskingSpanQueryBuilder extends QueryBuilder<FieldMaskingSpanQueryBuilder> implements SpanQueryBuilder,
+        BoostableQueryBuilder<FieldMaskingSpanQueryBuilder> {
 
     public static final String NAME = "field_masking_span";
 
-    private final SpanQueryBuilder queryBuilder;
+    private final QueryBuilder queryBuilder;
```

Comment:
I think I would change this back and rather cast in readFrom and barf if what we get is not a span query (not possible given that writeTo serialized a span query) ?